homeassistant/components/awair/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for Awair."""
+
+from typing import Union
+
+from python_awair import Awair
+from python_awair.exceptions import AuthError, AwairError
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, ConfigFlow
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, LOGGER  # pylint: disable=unused-import
+
+
+class AwairFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Awair."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    def __init__(self) -> None:
+        """Set up the Awair configuration flow."""
+        self._errors = {}
+
+    async def async_step_import(self, conf: dict):
+        """Import a configuration from config.yaml."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason="already_setup")
+
+        if not conf[CONF_ACCESS_TOKEN]:
+            return self.async_abort(reason="auth")
+
+        await self.__abort_if_configured(conf[CONF_ACCESS_TOKEN])
+
+        return self.async_create_entry(
```

Comment:
The connection should be checked before creating the entry also for the import step.

Reply:
I was working under the assumption that if there was an entry in someone's `configuration.yaml`, it was probably working already. But perhaps that's a foolish assumption 😆 

I'll update as suggested.